### DISCUSSION TITLE
Help with rule to seperate JSON without keys

### DISCUSSION THREAD

**Discussion by linesquarecube** [arendst/Tasmota#21077] [Q&A]
Hi everyone, 

I'm reading in serial data from my coffee machine which I want to separate in single values. I've read through the [rules documentation](https://tasmota.github.io/docs/Rules/#processing-json-received-from-softwareserialbridge) but I don't get how to separate values without a key. The values I get are just separated by a "," or even nothing.

Example data:
```
21:57:20.887 MQT: tele/tasmota_57F5ED/RESULT = {"SSerialReceived":"C1.06,110,112,091,0000,0,0\r\n"}
21:57:21.420 MQT: tele/tasmota_57F5ED/RESULT = {"SSerialReceived":"C1.06,110,112,091,0000,0,0\r\n"}
21:57:21.958 MQT: tele/tasmota_57F5ED/RESULT = {"SSerialReceived":"C1.06,110,112,091,0000,0,0\r\n"}
```
Example Data description: **C1.06,110,112,091,0000,1,0\n**
- The delimitter is: **,**
- The end of data is: **\n**
- Length: **26**

Data | Description
-- | --
C | Machine-Mode: C = CoffeeMode; V = Vapour/SteamMode
1.06 | Firmware
110 | Current Steam Temperature in Celsius
112 | Target Steam Temperature in Celsius
091 | Curent Hx Temperature in Celsius
0000 | Countdown Boost-Mode 
1 | Heat state (0 = off; 1= on)
0 | Pump state (0 = off; 1= on)

Maybe someone with more experience than me can help me solving my problem.

Thanks :)

**Comment by sfromis** [comment]
You can't do string parsing using rules.

To do it in Tasmota, my preference would be the Berry language, only available with ESP32.

**Reply by linesquarecube** [reply]
Thank you for answering this fast! As I'm using an ESP8266, I will try to do the parsing in Home Assistant.
